MyString.java

```diff
@@ -36,10 +48,29 @@ public static int countChar(String str, char ch) {
      * @return true is str1 is a subset of str2, false otherwise
      */
     public static boolean subsetOf(String str1, String str2) {
-         //// Replace the following statement with your code
-        return false;
+        if (str1.isEmpty() || str1.equals(str2)){
+            return true;
+        }
+
+        if (str1.length() > str2.length()){
+            return false;
+        } 
+        else 
+        {
+            for (int i = 0; i < str1.length(); i++){
+
+                if (countChar(str1, str1.charAt(i)) > countChar(str2, str1.charAt(i))){
+                    return false;
+                }
+                
+            }
+        }
+        return true;

```

Comment:
else is not needed , if it will enter in 1 of first 2 ifs it will exit the code..